tests/test_json_schema_to_grammar.py

```diff
@@ -0,0 +1,325 @@
+import json
+
+import pytest
+from jsonschema import validate
+
+from guidance._grammar import GrammarFunction
+from guidance._json_schema_to_grammar import json_schema_to_grammar
+from guidance._parser import EarleyCommitParser, ParserException
+
+
+def to_compact_json(target: any) -> str:
+    # See 'Compact Encoding':
+    # https://docs.python.org/3/library/json.html
+    # Since this is ultimately about the generated
+    # output, we don't need to worry about pretty printing
+    # and whitespace
+    return json.dumps(target, separators=(",", ":"))
+
+
+def check_string_with_grammar(input_string: str, grammar: GrammarFunction):
+    parser = EarleyCommitParser(grammar)
+
+    print(f"Checking {input_string}")
+    for c in input_string:
+        print(f"Working on: {c}")
+        print(f"Valid next bytes: {parser.valid_next_bytes()}")
+        next_byte = bytes(c, encoding="utf8")
+        print(f"Consuming: {next_byte}")
+        parser.consume_byte(next_byte)
```

Comment:
I think we should shift this logic over to the `GrammarFunction.match()` method -- I'd slowly like to use that method as the main check for grammar compliance across the codebase

Reply:
Done